src/views/open-api-docs-preview/components/open-api-preview-inputs/index.tsx

```diff
@@ -0,0 +1,149 @@
+// Third-party
+import { useEffect, useState } from 'react'
+import classNames from 'classnames'
+// Components
+import Button from 'components/button'
+import InlineAlert from 'components/inline-alert'
+// Inputs
+import { FileStringInput } from '../file-string-input'
+import { TextareaInput } from '../textarea-input'
+// Utils
+import { fetchOpenApiStaticProps } from './utils/fetch-open-api-static-props'
+// Types
+import type { OpenApiDocsViewProps } from 'views/open-api-docs-view/types'
+// Styles
+import s from './open-api-preview-inputs.module.css'
+
+interface InputValues {
+	openApiJsonString: string
+	openApiDescription: string
+}
+
+/**
+ * Render a fixed panel container with inputs that allow control over the
+ * static props for `OpenApiDocsView`.
+ */
+export function OpenApiPreviewInputs({
+	setStaticProps,
+}: {
+	setStaticProps: (v: OpenApiDocsViewProps) => void
+}) {
+	const [error, setError] = useState<{
+		title: string
+		description: string
+		error: string
+	}>()
+	const [isLoading, setIsLoading] = useState(false)
```

Comment:
This is non-blocking suggestion to reduce the sheer # of  lines of code, and prop passing:

The `isLoading`,`error`, and data fetching code in this component can be conveniently abstracted away by a react-query `useQuery` hook... (and a fixed **querykey**; see more below)

Similarly, the lone `setStaticProps` state setter could be completely removed. The parent that owns `[staticProps, setStaticProps]` could replace `useState` and instead do a "read-only" usage of `useQuery` with the previously mentioned fixed **querykey**.

Reply:
Ah neat - would love to explore this in a future PR! I tried for a bit, but ran into some complications around having the query only fetch initially (and only refetch) when the `updateStaticProps` function is called - wasn't immediately clear based on the docs the best way to do that. Very open to refactoring this later 🙇‍♂️